### DISCUSSION TITLE
How do I read a file in vercel?

### DISCUSSION THREAD

**Discussion by therealsujitk** [vercel/vercel#5383] [Help] [answered]
My index.js file has this code:
```js
var rawData = fs.readFileSync('details.json', (err, data) => {
    if (err) throw err;
});
var details = JSON.parse(rawData);
var versionName = details['version-name'];
```
It works perfectly on localhost but on vercel it gives a bad gateway error. When I replace `'details.json'` with `'./details.json'` it gives internal server error. Am I supposed to add something to **vercel.json** for this to work?

**Comment by mcsdevv** [comment, marked as answer]
Hello @therealsujitk, please take a look at this article: https://vercel.com/knowledge/how-can-i-use-files-in-serverless-functions

**Comment by theapplefolks** [comment]
How do you get the pattern for the function name to work with catch-all route names?
